File: trustly-android-sdk/src/test/java/net/trustly/android/sdk/interfaces/TrustlyCallbackTest.java
package net.trustly.android.sdk.interfaces;

import static org.mockito.Mockito.clearInvocations;
import static org.mockito.Mockito.verify;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashMap;

public class TrustlyCallbackTest {

    @Mock
    public TrustlyCallback<Trustly, HashMap<String, String>> mockTrustlyCallback;

    @Mock
    public Trustly mockTrustly;

    @Before
    public void setUp() {
        MockitoAnnotations.openMocks(this);
    }

    @After
    public void tearDown() {
        clearInvocations(mockTrustlyCallback, mockTrustly);
    }

    @Test
    public void shouldValidateTrustlyCallbackWhenEventAndParametersArePassed() {
        HashMap<String, String> parameters = new HashMap<>();
        parameters.put("param1", "param1");
        parameters.put("param2", "param2");
        new TrustlyCallbackImpl(mockTrustlyCallback).handleParameters(mockTrustly, parameters);
        verify(mockTrustlyCallback).handle(mockTrustly, parameters);
    }

    @Test
    public void shouldValidateTrustlyCallbackWhenEventAndNullParametersArePassed() {
        new TrustlyCallbackImpl(mockTrustlyCallback).handleParameters(mockTrustly, null);
        verify(mockTrustlyCallback).handle(mockTrustly, null);
    }

    private static class TrustlyCallbackImpl {

        private final TrustlyCallback<Trustly, HashMap<String, String>> trustlyCallback;

        TrustlyCallbackImpl(TrustlyCallback<Trustly, HashMap<String, String>> trustlyCallback) {
            this.trustlyCallback = trustlyCallback;
        }

        void handleParameters(Trustly trustly, HashMap<String, String> parameters) {
            trustlyCallback.handle(trustly, parameters);
        }

    }

}

File: trustly-android-sdk/src/test/java/net/trustly/android/sdk/interfaces/TrustlyCallbackTest.kt
package net.trustly.android.sdk.interfaces

import org.junit.After
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.clearInvocations
import org.mockito.Mockito.verify
import org.mockito.MockitoAnnotations

class TrustlyCallbackTest {

    @Mock
    private lateinit var mockTrustlyCallback: TrustlyCallback<Trustly?, HashMap<String, String>?>

    @Mock
    private lateinit var mockTrustly: Trustly

    @Before
    fun setUp() {
        MockitoAnnotations.openMocks(this)
    }

    @After
    fun tearDown() {
        clearInvocations(mockTrustlyCallback, mockTrustly)
    }

    @Test
    fun shouldValidateTrustlyCallbackWhenEventAndParametersArePassed() {
        val parameters = hashMapOf(
            "key1" to "value1",
            "key2" to "value2"
        )

        TrustlyCallbackImpl(mockTrustlyCallback).handleParameters(mockTrustly, parameters)
        verify(mockTrustlyCallback).handle(mockTrustly, parameters)
    }

    @Test
    fun shouldValidateTrustlyCallbackWhenEventAndNullParametersArePassed() {
        TrustlyCallbackImpl(mockTrustlyCallback).handleParameters(mockTrustly, null)
        verify(mockTrustlyCallback).handle(mockTrustly, null)
    }

    private class TrustlyCallbackImpl(private val trustlyCallback: TrustlyCallback<Trustly?, HashMap<String, String>?>) {

        fun handleParameters(trustly: Trustly, parameters: HashMap<String, String>?) {
            trustlyCallback.handle(trustly, parameters)
        }

    }

}